owncloudApp/src/androidTest/java/com/owncloud/android/settings/more/SettingsFragmentMoreSectionTest.kt

```diff
@@ -160,133 +149,127 @@ class SettingsFragmentMoreSectionTest {
-        assertEquals(PREFERENCE_MORE_CATEGORY, prefMoreCategory.key)
-        assertEquals(context.getString(R.string.prefs_category_more), prefMoreCategory.title)
-        assertNull(prefMoreCategory.summary)
-        assertTrue(prefMoreCategory.isVisible)
-
-        onView(withText(R.string.prefs_help)).check(matches(isDisplayed()))
-        assertEquals(PREFERENCE_HELP, prefHelp?.key)
-        assertEquals(context.getString(R.string.prefs_help), prefHelp?.title)
-        assertNull(prefHelp?.summary)
-        assertTrue(prefHelp?.isVisible == true)
-        assertTrue(prefHelp?.isEnabled == true)
-
-        onView(withText(R.string.prefs_sync_calendar_contacts)).check(matches(isDisplayed()))
-        onView(withText(R.string.prefs_sync_calendar_contacts_summary)).check(matches(isDisplayed()))
-        assertEquals(PREFERENCE_SYNC_CALENDAR_CONTACTS, prefSync?.key)
-        assertEquals(context.getString(R.string.prefs_sync_calendar_contacts), prefSync?.title)
-        assertEquals(context.getString(R.string.prefs_sync_calendar_contacts_summary), prefSync?.summary)
-        assertTrue(prefSync?.isVisible == true)
-        assertTrue(prefSync?.isEnabled == true)
-
-        onView(withText(R.string.prefs_recommend)).check(matches(isDisplayed()))
-        assertEquals(PREFERENCE_RECOMMEND, prefRecommend?.key)
-        assertEquals(context.getString(R.string.prefs_recommend), prefRecommend?.title)
-        assertNull(prefRecommend?.summary)
-        assertTrue(prefRecommend?.isVisible == true)
-        assertTrue(prefRecommend?.isEnabled == true)
-
-        onView(withText(R.string.prefs_feedback)).check(matches(isDisplayed()))
-        assertEquals(PREFERENCE_FEEDBACK, prefFeedback?.key)
-        assertEquals(context.getString(R.string.prefs_feedback), prefFeedback?.title)
-        assertNull(prefFeedback?.summary)
-        assertTrue(prefFeedback?.isVisible == true)
-        assertTrue(prefFeedback?.isEnabled == true)
-
-        onView(withText(R.string.prefs_privacy_policy)).check(matches(isDisplayed()))
-        assertEquals(PREFERENCE_PRIVACY_POLICY, prefPrivacyPolicy?.key)
-        assertEquals(context.getString(R.string.prefs_privacy_policy), prefPrivacyPolicy?.title)
-        assertNull(prefPrivacyPolicy?.summary)
-        assertTrue(prefPrivacyPolicy?.isVisible == true)
-        assertTrue(prefPrivacyPolicy?.isEnabled == true)
-
-        onView(withText(R.string.prefs_imprint)).check(matches(isDisplayed()))
-        assertEquals(PREFERENCE_IMPRINT, prefImprint?.key)
-        assertEquals(context.getString(R.string.prefs_imprint), prefImprint?.title)
-        assertNull(prefImprint?.summary)
-        assertTrue(prefImprint?.isVisible == true)
+        prefMoreCategory.verifyPreference(
+            keyPref = PREFERENCE_MORE_CATEGORY,
+            titlePref = context.getString(R.string.prefs_category_more),
+            visible = true
+        )
+
+        prefHelp = getPreference(PREFERENCE_HELP)
+        prefHelp?.verifyPreference(
```

Comment:
Nice! But just a comment. If we use that `?`, and the preference is null, it won't be verified. If it is null, we probably want the test to fail instead of leaving it without verification. What do you think?

Reply:
Well the idea here is that in that specific test, `moreView`, no preferences are removed (and consequently all of them are visible). Then we have for each brandable preference a test checking it is null if we don't set it to true in `launchTest()`. But it's ok, for more security and to avoid unexpected cases, I will include `assertNotNull()` statements in this test